Cabal/tests/PackageTests/PackageTester.hs

```diff
@@ -741,6 +746,18 @@ ghcFileModDelay = do
               = mtimeChangeDelay suite
     liftIO $ threadDelay delay
 
+-- | Create a symlink if necessary. Does not work on Windows.
+createSymlink :: FilePath -> FilePath -> TestM ()
+#ifdef mingw32_HOST_OS
+createSymlink _oldpath _newpath =
+  error "PackageTests.PackageTester.createSymlink: does not work on Windows!"
+#else
+createSymlink oldpath newpath = do
+  symlinkExists <- liftIO $ doesFileExist newpath
```

Comment:
I'm pretty sure you want to delete the old symlink if it exists, otherwise a stale symlink will stick around and cause problems.